### PR TITLE
Check fixed choices in name collision checker

### PR DESCRIPTION
**Pull request by sofiafaro-da**
This is a follow up to #11364. This PR checks that the list of inherited
choice names is correct in the type checker, and moves the "fixed choice name
collision" check into the name collision checker, since it's now a local check.

Haskell side only for now. Part of #11137.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs [hunk 1]
```diff
@@ -885,27 +885,18 @@ checkTemplate m t@(Template _loc tpl param precond signatories observers text ch
...
-
 checkIfaceImplementation :: MonadGamma m => Qualified TypeConName -> TemplateImplements -> m ()
 checkIfaceImplementation tplTcon TemplateImplements{..} = do
   let tplName = qualObject tplTcon
-  DefInterface {intVirtualChoices, intMethods} <- inWorld $ lookupInterface tpiInterface
+  DefInterface {intFixedChoices, intVirtualChoices, intMethods} <- inWorld $ lookupInterface tpiInterface
+
+  -- check fixed choices
+  let inheritedChoices = S.fromList (NM.names intFixedChoices)
+  unless (inheritedChoices == tpiInheritedChoiceNames) $
```
Comment (cocreature) [inline compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs:898]: is it possible to trigger this in Daml? If so, a test might be nice.

File: compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs [hunk 2]
```diff
@@ -929,6 +920,7 @@ checkIfaceImplementation tplTcon TemplateImplements{..} = do
       Just InterfaceMethod{ifmType} ->
         checkExpr tpiMethodExpr (TCon tplTcon :-> ifmType)
 
+
```
Comment (cocreature) [inline compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs:923]:
```suggestion
```
unintentional? if not ignore

**Review by cocreature — APPROVED** [review]
Thanks!

File: compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs (hunk 1 above)
Comment (sofiafaro-da) [reply compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs:898]: I don't think it's possible. ATM this only serves as a sanity check against the LF completer.

File: compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs (hunk 2 above)
Comment (sofiafaro-da) [reply compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs:923]: unintentional!

**Pull request merged**